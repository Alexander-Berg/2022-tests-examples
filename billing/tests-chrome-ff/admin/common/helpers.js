const oebsStatusSelector = '.src-common-components-Person-___person-module__person-row';
const typeIdSelector = '.src-common-components-Person-___person-module__person-id__type-id';
const personName = '.yb-person-id__name';

module.exports.hideElements = [oebsStatusSelector, typeIdSelector];
module.exports.ignoreElements = [personName];
module.exports.personModalSelector = '.yb-person-modal';
