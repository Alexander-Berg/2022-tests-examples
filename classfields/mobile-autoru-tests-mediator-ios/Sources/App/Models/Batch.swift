import Mediation
import Vapor

struct Batch: Content {
    let id: String
    let tests: [String]
    let parallelize: Bool
    let executionInfo: ExecutionInfo?

    struct ExecutionInfo: Content {
        let dequeueTime: Date
        let completeTime: Date?
        let executor: TeamcityBuildID
    }

    init(from batch: TestsBatch) {
        self.id = batch.id
        self.tests = batch.tests.map(\.name)
        self.parallelize = batch.parallelize
        self.executionInfo = batch.executionInfo.flatMap {
            ExecutionInfo(dequeueTime: $0.dequeueTime, completeTime: $0.completeTime, executor: $0.executor)
        }
    }
}
