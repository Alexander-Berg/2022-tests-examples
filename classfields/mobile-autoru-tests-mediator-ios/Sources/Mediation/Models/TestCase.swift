import Foundation

/// Описание теста, которое подается на вход (из аппа)
public struct TestCase: Hashable {
    public let name: String

    public init(_ name: String) {
        self.name = name
    }
}

extension TestCase: ExpressibleByStringLiteral {
    public init(stringLiteral value: String) {
        self.init(value)
    }
}
