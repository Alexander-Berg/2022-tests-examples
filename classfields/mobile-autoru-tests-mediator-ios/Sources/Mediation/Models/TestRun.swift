import Foundation

public typealias TeamcityBuildID = Int
public typealias TeamcityBuildNumber = String
public typealias BuildConfigurationID = String

/// Описания одного прогона тестов
public final class TestRun {
    let id: String
    let initiatorBuildID: TeamcityBuildID
    let initiatorBuildNumber: TeamcityBuildNumber
    let sourceBranch: String
    let sourcePullRequest: Int?
    let sourceBuildID: TeamcityBuildID

    let testsRunResultConfigID: BuildConfigurationID

    private(set) var batches: [TestsBatch]
    let strategy: SplitStrategy
    let maxRetriesCount: Int

    private var resultsMergeBuildID: TeamcityBuildID?

    private(set) var activeBuilds: Set<TeamcityBuildID>
    private(set) var completedBuilds: Set<TeamcityBuildID>
    private(set) var failedBuilds: Set<TeamcityBuildID>

    private var retriesCount: [TestCase: Int]

    var completedBuildsWithAtLeastOneBatch: Set<TeamcityBuildID> {
        let batchesExecutors = Set(batches.compactMap(\.executionInfo?.executor))
        return completedBuilds.intersection(batchesExecutors)
    }

    init(
        initiatorBuildID: TeamcityBuildID,
        initiatorBuildNumber: TeamcityBuildNumber,
        sourceBranch: String,
        sourcePullRequest: Int?,
        sourceBuildID: TeamcityBuildID,
        tests: [TestCase],
        strategy: SplitStrategy,
        maxRetriesCount: Int,
        testsRunResultConfigID: BuildConfigurationID
    ) {
        self.id = UUID().uuidString
        self.initiatorBuildID = initiatorBuildID
        self.initiatorBuildNumber = initiatorBuildNumber
        self.sourceBranch = sourceBranch
        self.sourcePullRequest = sourcePullRequest
        self.sourceBuildID = sourceBuildID
        self.batches = strategy.split(tests: tests)
        self.strategy = strategy
        self.maxRetriesCount = maxRetriesCount

        self.activeBuilds = .init()
        self.completedBuilds = .init()
        self.failedBuilds = .init()

        self.retriesCount = [:]
        for test in tests {
            self.retriesCount[test] = 0
        }

        self.testsRunResultConfigID = testsRunResultConfigID
    }

    func setActiveBuilds(_ builds: [TeamcityBuildID]) {
        activeBuilds = Set(builds)
    }

    func markBuildAsCompleted(id: TeamcityBuildID) throws -> Bool {
        guard activeBuilds.contains(id) else { throw Error.inconsistentExecution }

        activeBuilds.remove(id)
        completedBuilds.insert(id)

        if activeBuilds.isEmpty {
            return true
        }

        return false
    }

    func setResultsMerge(buildID: TeamcityBuildID) {
        resultsMergeBuildID = buildID
    }

    func retry(tests: [TestCase]) {
        var filteredTests: [TestCase] = []

        for test in tests {
            retriesCount[test, default: 0] += 1

            if retriesCount[test, default: 0] <= maxRetriesCount {
                filteredTests.append(test)
            }
        }

        var currentBatches: [TestsBatch] = []
        var waitingTests: [TestCase] = []

        for batch in batches {
            if batch.alreadyExecuting {
                currentBatches.append(batch)
            } else {
                waitingTests.append(contentsOf: batch.tests)
            }
        }

        waitingTests.append(contentsOf: filteredTests)

        let newBatches = strategy.split(tests: waitingTests)
        batches = currentBatches + newBatches
    }

    enum Error: Swift.Error {
        case inconsistentExecution
    }
}
