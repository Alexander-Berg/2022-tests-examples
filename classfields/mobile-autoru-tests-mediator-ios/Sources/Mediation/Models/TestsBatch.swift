import Foundation

/// Описание одной задачи на прогон тестов
public final class TestsBatch {
    public let id: String
    public let tests: [TestCase]
    private(set) public var executionInfo: ExecutionInfo?
    public let parallelize: Bool

    var alreadyExecuting: Bool { executionInfo != nil }

    var isCompleted: Bool { executionInfo?.completeTime != nil }

    init(tests: [TestCase], parallelize: Bool = true) {
        self.id = UUID().uuidString
        self.tests = tests
        self.parallelize = parallelize
    }

    func markAsExecuting(buildID: TeamcityBuildID) throws {
        if alreadyExecuting { throw Error.inconsistentExecution }

        executionInfo = ExecutionInfo(
            executor: buildID,
            dequeueTime: Date(),
            completeTime: nil
        )
    }

    func markAsCompleted() throws {
        if !alreadyExecuting { throw Error.inconsistentExecution }

        executionInfo?.completeTime = Date()
    }

    func resetExecutionInfo() {
        executionInfo = nil
    }

    enum Error: Swift.Error {
        case inconsistentExecution
    }
}

extension TestsBatch {
    public struct ExecutionInfo {
        public let executor: TeamcityBuildID
        public let dequeueTime: Date
        public var completeTime: Date?
    }
}
