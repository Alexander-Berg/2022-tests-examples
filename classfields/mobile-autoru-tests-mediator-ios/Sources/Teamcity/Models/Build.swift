import Foundation

public struct Build: Decodable {
    public let id: Int
    public let number: String?
    public let state: State
    public let buildTypeId: BuildConfigurationID

    public init(
        id: Int,
        number: String,
        state: State,
        buildTypeId: BuildConfigurationID
    ) {
        self.id = id
        self.number = number
        self.state = state
        self.buildTypeId = buildTypeId
    }
}

extension Build {
    public enum State: String, Decodable {
        case finished
        case running
        case queued
        case unknown
    }

    public enum Status: String, Decodable {
        case status = "STATUS"
        case failure = "FAILURE"
        case unknown = "UNKNOWN"
    }
}
