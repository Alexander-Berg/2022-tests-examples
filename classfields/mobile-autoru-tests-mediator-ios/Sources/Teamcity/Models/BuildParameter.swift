import Foundation

public enum BuildParameter: String {
    case initiatorBuildID = "initiator_build_id"
    case initiatorBuildNumber = "initiator_build_number"
    case sourceBuildID = "source_build_id"
    case sourceBranchName = "source_branch_name"
    case sourcePullRequestID = "source_pr_id"
    case finishedBuildsIDs = "finished_builds_ids"
    case mediatorBuildUUID = "mediator_build_uuid"
    case mediatorRunUUID = "mediator_run_uuid"

    public enum Value {
        case string(String)
        case int(Int)
        indirect case array([Value])

        var value: String {
            switch self {
            case .string(let value):
                return value
            case .int(let value):
                return "\(value)"
            case .array(let value):
                return value.map(\.value).joined(separator: ",")
            }
        }
    }
}
