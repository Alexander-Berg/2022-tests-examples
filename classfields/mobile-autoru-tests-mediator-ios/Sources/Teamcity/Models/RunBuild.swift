import Foundation

struct RunBuild: Codable {
    let branchName: String
    let buildType: BuildType
    let comment: Comment?
    let properties: Properties

    struct BuildType: Codable {
        let id: String
    }

    struct Comment: Codable {
        let text: String
    }

    struct Properties: Codable {
        let property: [Property]

        struct Property: Codable {
            let name: String
            let value: String
        }
    }
}
