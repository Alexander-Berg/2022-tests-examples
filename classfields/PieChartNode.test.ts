import type { YogaNode } from 'app/server/tmpl/YogaNode';
import { PaletteColor } from 'app/server/tmpl/android/v1/helpers/Palette';
import { spreadBackgroundColor, spreadTextColor } from 'app/server/tmpl/android/v1/models/Models';
import { yogaToXml } from 'app/server/tmpl/XMLBuilder';


describe('pie chart', () => {
    it('node', () => {
        const ratingNode: YogaNode = {
            name: 'stack',
            id: 'pie_chart_node',
            flexDirection: 'column',
            ...spreadBackgroundColor(PaletteColor.surface),
            children: [
                {
                    name: 'frame',
                    width: 50,
                    height: 50,
                    children: [
                        {
                            name: 'pieGraph',
                            id: 'pie_1',
                        },
                        {
                            name: 'stack',
                            width: 50,
                            height: 50,
                            alignItems: 'center',
                            justifyContent: 'center',
                            children: [
                                {
                                    name: 'text',
                                    ...spreadTextColor(PaletteColor.onSurfaceEmphasisHigh),
                                    text: '70',
                                },
                            ],
                        },
                    ],
                },
                {
                    name: 'pieGraph',
                    id: 'pie_2',
                    width: 50,
                    height: 50,
                },
            ],
        };
        const xml = yogaToXml([ ratingNode ]);
        expect(xml).toMatchSnapshot();
    });
});
