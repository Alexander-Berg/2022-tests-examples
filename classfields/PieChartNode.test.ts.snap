// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pie chart node 1`] = `"<list><stack id=\\"pie_chart_node\\" flexDirection=\\"column\\" backgroundColor=\\"FFFFFF\\" backgroundColorDark=\\"1C1C1E\\" backgroundColorID=\\"45\\"><frame width=\\"50\\" height=\\"50\\"><pieGraph id=\\"pie_1\\"/><stack width=\\"50\\" height=\\"50\\" alignItems=\\"center\\" justifyContent=\\"center\\"><text textColor=\\"000000\\" textColorDark=\\"FFFFFF\\" textColorID=\\"47\\" text=\\"70\\"/></stack></frame><pieGraph id=\\"pie_2\\" width=\\"50\\" height=\\"50\\"/></stack></list>"`;
